### PR TITLE
[Tox Support] Additional Changes for `tox` default environment name for type checking and flake8 configuration

### PR DESCRIPTION
**Pull request by R7L208**
# Default Environment Change
- updating the `type` environment to `type-check` which matches the actual environment configuration
- the default environment list wasn't updated when renaming the env to `type-check`

this is needed because `tox list` is showing the `type` default environment which does not exist and `type-check` as an additional environment.

# flake8 Config Change
- flake8 config was defined in `.flake8` and `tox.ini`. The configurations in `tox.ini` were incomplete and took precedence when running `flake8` natively and the linting environment `tox -e lint`.
- removing the `[flake8]` config section from `tox.ini` and migrating all config rules to `.flake8` forces `tox` to look in `.flake8` for config rules and does not break the current GH actions

### CODE REVIEW

File: python/.flake8 [hunk 1]
```diff
@@ -11,8 +11,18 @@ extend-ignore =
     # 'from module import *' used; unable to detect undefined names (F403)
     F403
```
Comment (tnixon) [inline python/.flake8:12]: Is this really required? This type of import is bad form, I'd like to refactor it if we can

Comment (tnixon) [reply python/.flake8:12]: Do we know which file fails this check?

Comment (R7L208) [reply python/.flake8]: Removed the config and violations

**Review by tnixon — APPROVED** [review]
LGTM

**Pull request merged**